### DISCUSSION TITLE
Signed wordlists

### DISCUSSION THREAD

**Discussion by christianwengert** [keepassxreboot/keepassxc#11050] [Ideas]
Hello

I've seen that KeePassXC will read PGP signed wordlists (but not verifying the signature). 

In principle, signing these files seems like a good idea because the integrity of said wordlists can be protected. 

I would be interested in adding this to KeePassXC but I have a couple of questions before:

1. Who will be signing the wordlists? Who has the private key? Would that be part of dev team of KeePassXC?
2. Should - if the maintainers keep the key - KeePassXC manage the wordlists inside the repository (of many languages)?
3. Could it be an option to use a different signing mechanisms than PGP? Re-implementing PGP seems quite a. lot of work (using the botan library and not adding another dependency). 

I think we could implement a mechanism that is more inspired by `openssl` (something like `openssl dgst -sha256 -sign privatekey.pem -out data.txt.signature data.txt`). This would be much simpler to implement in botan but would lead to two separate files (the wordlist and the wordlist.signature).

Again, the principal question is: Should such a mechanism (openssl or PGP) be added to KeePassXC and where will we place the wordlists (ideally multilingual) and the signatures and who will have the private key?

Thanks for your thoughts.

**Comment by phoerious** [comment]
I don't see any advantage of signing word lists.

**Comment by droidmonkey** [comment]
> Again, the principal question is: Should such a mechanism (openssl or PGP) be added to KeePassXC

No,  that is a ton of overhead for no gain in security

**Comment by christianwengert** [comment]
Ok thanks for your responses. Indeed the overhead is rather big. 

The little gain in security would be for the case where an attacker would be able to change your wordlist such that the generated passphrases would have much less entropy. 
I can definitely live with not implementing this part :)

Also if https://github.com/keepassxreboot/keepassxc/pull/10995 gets accepted, there is no more small entropy wordlists anyway :)
